src/utils/dot-env.js

```diff
@@ -6,9 +6,8 @@ const { isFileAsync, readFileAsync } = require('../lib/fs')
 
 const { warn: warn_ } = require('./command-helpers')
 
-const loadDotEnvFiles = async function ({ projectDir, warnLog }) {
-  // a stub utility is used in tests
-  const warn = warnLog || warn_
+// a stub utility is used in tests for warn
+const loadDotEnvFiles = async function ({ projectDir, warn = warn_ }) {
```

Comment:
Should the argument `warn` be removed altogether?

It seems like the reason this is here is because `dot-env.test.js` is mocking `warn()`. This is probably to remove warning messages printed during tests. However, mocking seems the wrong way to solve this, as it would not scale well: many test files would need to mock `warn()`. A better way IMHO would be to either:
  - Mock the global `console` object
  - Spawn a child process without forwarding its `stdout` or `stderr`

Based on this, what are your thoughts on the following?
  - Removing the `warnLog`/`warn` argument from this function.
  - Import the `warn` utility directly instead
  - In `dot-env.test.js`, do not pass any `warn` stub

Reply:
I'd definitely like to remove the `warn` argument. 

Internally in `@oclif/errors`, `warn` relies on `console.error` for logging but I don't think we should rely on that to change the global object.

Also, AFAIK in some places the logs are tested which means that we can't use the test environment flag to disable them.
(But, we could use an alternate environment variable which can be set before tests that require logging to be disabled and removed during cleanup.)

> In dot-env.test.js, do not pass any warn stub

I went through the tests and they only seem to be checking the results. Why was `warn` being suppressed in that case?

Reply:
> Why was warn being suppressed in that case?

My guess is that this was to remove the verbosity of the console output when running tests. However, splitting the logging logic to a parent function would allow the tests to import the core logic without printing any log messages.